libc/newhdrgen/class_implementation/classes/function.py

```diff
@@ -23,9 +23,12 @@ def __init__(
         self.attributes = attributes or ""
 
     def __str__(self):
-        attributes_str = self.attributes
+        attributes_str = " ".join(self.attributes)
         arguments_str = ", ".join(self.arguments)
-        result = f"{self.return_type} {self.name}({arguments_str}){attributes_str};"
+        if attributes_str == "":
```

Comment:
Do we need the if/else here or can we just put {attributes_str} in the front either way?

Reply:
That's what I wanted to do, but if we have attributes we will need a space between the attribute and the return type. Normally there would be no space in front.